File: Auto-Parts-Management-insurance_form/Server/parthub-spring-boot-server/src/main/java/com/xoftex/parthub/models/RepairMenu/RepairMenuResponse.java
package com.xoftex.parthub.models.RepairMenu;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class RepairMenuResponse {
    @SerializedName("repair_menu")
    public RepairMenu repairMenu;

    public static class RepairMenu {
        @SerializedName("repair_request")
        public String repairRequest;

        public List<Section> sections;
    }

    public static class Section {
        public String title;

        // For flexibility, use Object here and cast as needed
        public List<Object> items;
    }

    // Used for sections with standard repair tasks
    public static class RepairTask {
        public String description;
        public String details; // optional
        @SerializedName("labor_hours")
        public Double laborHours;
    }

    // Used for fluid section
    public static class FluidItem {
        public String type;
        public String spec;
        @SerializedName("quantity_quarts")
        public Double quantityQuarts;
        @SerializedName("quantity_gallons")
        public Double quantityGallons;
        @SerializedName("part_number")
        public String partNumber;
    }

    // Used for estimate summary section
    public static class EstimateItem {
        public String item;
        @SerializedName("estimated_cost_usd_fron")
        public Double estimatedCostUsdFrom;
        @SerializedName("estimated_cost_usd_to")
        public Double estimatedCostUsdTo;
    }
}

File: Auto-Parts-Management-insurance_form/Client/angular-parthub/src/app/models/repair.menu.response.ts
export class RepairMenuResponse {
    repairMenu: RepairMenu = new RepairMenu();
}

export class RepairMenu {
    repairRequest: string = "";
    sections: Section[] = new Array();
}

export class Section {
    title: string = "";
    items: (RepairTask | FluidItem | EstimateItem)[] = new Array();
}

export class RepairTask {
    description: string = "";
    type: string = "";
    item: string = "";
    spec?: string = "";
    details?: string = "";
    labor_hours: number = 0;
    estimated_cost_usd_fron: number = 0;
    estimated_cost_usd_to: number = 0;
}

export class FluidItem {
    type: string = "";
    item: string = "";
    spec?: string = "";
    details?: string = "";
    description: string = "";
    quantity_quarts?: number;
    quantity_gallons?: number;
    labor_hours: number = 0;
    part_number: string = "";
    estimated_cost_usd_fron: number = 0;
    estimated_cost_usd_to: number = 0;
}

export class EstimateItem {
    type: string = "";
    item: string = "";
    spec?: string = "";
    description: string = "";
    labor_hours: number = 0;
    details?: string = "";
    estimated_cost_usd_fron: number = 0;
    estimated_cost_usd_to: number = 0;
}
